lib/controllers/helpers/passport.js

```diff
@@ -0,0 +1,55 @@
+"use strict";
+const passport = require("passport");
+const mongoose = require("mongoose");
+const config = require("../../../config");
+const JwtStrategy = require("passport-jwt").Strategy;
+const ExtractJwt = require("passport-jwt").ExtractJwt;
+const LocalStrategy = require("passport-local");
+
+const User = mongoose.model("User");
+// Create local strategy
+const localOptions = { usernameField: "email" };
+const localLogin = new LocalStrategy(localOptions, function(email, password, done) {
+  // Verify this email and password, call done with the user
+  // if it is the correct email and password
+  // otherwise, call done with false
+  User.findOne({ email: email }, function(err, user) {
+    if (err) { return done(err); }
+    if (!user) { return done(null, false); }
+
+    // compare passwords - is `password` equal to user.password?
+    user.comparePassword(password, function(err, isMatch) {
+      if (err) { return done(err); }
+      if (!isMatch) { return done(null, false); }
+
+      return done(null, user);
+    });
+  });
+});
+
+// Setup options for JWT Strategy
+const jwtOptions = {
+  jwtFromRequest: ExtractJwt.fromHeader("authorization"),
+  secretOrKey: config.jwtSecret
+};
+
+// Create JWT strategy
+const jwtLogin = new JwtStrategy(jwtOptions, function(payload, done) {
```

Comment:
It might be helpful to also pass in the request into this function. So:
`function(req, payload, done)` perhaps.